tests/specs/mcd/verification.k

```diff
@@ -138,6 +157,13 @@ module LEMMAS-MCD-HASKELL [symbolic, kore]
     rule #range(M:Memory [ K <- V ] [ K' := VS':ByteArray ], START, WIDTH) => #range(M [ K := (V : VS')        ], START, WIDTH)  requires K' +Int #sizeByteArray(VS') ==Int START +Int WIDTH andBool START <=Int K andBool 0 <Int WIDTH andBool 0 <=Int START
                                                                                                                                   andBool K +Int 1                    ==Int K'                                                                                [concrete(K, K', V, VS'), simplification]
 
+    rule { (M:Map [K <- V1]) #Equals (M [K <- V2]) } => { V1 #Equals V2 }
+         [simplification]
+
+    rule { M:Map [K <- V1] #Equals M } =>
+         { true #Equals K in_keys(M) andBool V1 ==K #lookup(M, K) }
+         [simplification]
```

Comment:
```suggestion
    rule { M:Map [K <- V1] #Equals M } =>
         { true #Equals K in_keys(M) andBool V1 ==K M[K] }
         [simplification]
```

Reply:
```suggestion
    rule { M:Map [K <- V1] #Equals M } =>
         { true #Equals K in_keys(M) andBool V1 ==K #lookup(M, K) }
       requires K in_keys(M) andBool isInt(M[K]) andBool #rangeUInt(256, M[K])
         [simplification]
```

Reply:
Need to eliminate the `#lookup.notInt` case and the `#lookup.none` case, and make sure we are in the `#lookup.some` case if we want to use `#lookup` on RHS here. https://github.com/runtimeverification/evm-semantics/blob/master/evm-types.md#storagememory-lookup